pilot/pkg/networking/core/v1alpha3/envoyfilter/listener_patch.go

```diff
@@ -124,36 +133,47 @@ func doListenerOperation(patchContext networking.EnvoyFilter_PatchContext,
 			proto.Merge(listener, cp.Value)
 		}
 	}
-
-	doFilterChainListOperation(patchContext, patches, listener)
+	if !applied {
```

Comment:
We should wrap this branch into the `monitoring.IncrementEnvoyFilterMetric`. I am thing something like `IncrementEnvoyFilterMetric(type, incBy int, patchError bool, isApplied bool) {}`. And inside the function you can just call `record(incBy)`, instead of `for skipped-- {incrementSkipped()}`. At the monitoring package, `increment` calls `record(1)` underlysing: https://github.com/istio/pkg/blob/482f0e4f1befc6a622203cf4157705a4998936fe/monitoring/monitoring.go#L189

Reply:
I was thinking about it but I did not like passing multiple flags with same values all the time like `1, false, applied `to many of the calls. I split them in to multiple functions. Please take a look.